sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -216,24 +236,28 @@ case class GpuCast(
         castDecimalToDecimal(input.getBase, from, to)
 
       case _ =>
-        doColumnar(input.getBase, input.dataType())
+        recursiveDoColumnar(input.getBase, input.dataType())
     }
   }
 
-  def doColumnar(input: ColumnView, sparkType: DataType): ColumnVector = {
-    (sparkType, dataType) match {
+  private def recursiveDoColumnar(
+                                   input: ColumnView,
+                                   fromDataType: DataType,
+                                   toDataType: DataType = dataType
+  ): ColumnVector = {
```

Comment:
Nit: Similarly odd indentation and return type placement that isn't consistent with other examples.